src/RecordComponent.cpp

```diff
@@ -350,7 +362,11 @@ void RecordComponent::readBase()
         val.has_value())
         setUnitSI(val.value());
     else
-        throw std::runtime_error("Unexpected Attribute datatype for 'unitSI'");
+        throw error::ReadError(
+            error::AffectedObject::Attribute,
+            error::Reason::UnexpectedContent,
+            {},
+            "Unexpected Attribute datatype for 'unitSI'");
```

Comment:
For such errors, maybe we should write also what we expect and what we got in the error message?
```suggestion
            "Unexpected Attribute datatype for 'unitSI' (expected: double; found: ...)");
```